src/auth/auth.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const path = require('path');
+const passport = require(path.resolve('src/auth/passport'));
+const jwt = require('jsonwebtoken');
+
+// login drives the workflow for a login strategy to be used and ensures
+// that a valid token is returned in the response
+exports.login = function (req, res, next) {
```

Comment:
Why wrap passport.authenticate in a function that takes the arguments and just passes it into passport.authenticate? Wouldn't it be the same as just

`
exports.login = passport.authenticate('local', { session: false }, function (err, user) {
    if (err) {
      return res.status(500).json({ message: `500 ${err.message}` });
    }
    if (!user) {
      return res.status(401).json({ message: '401 credentials are invalid' });
    }

    // Replacement for serialising the returned token
    const token = jwt.sign({ userId: user.id }, process.env.EXPRESS_SESSION_SECRET);
    res.json({ token });
  `

Reply:
no `res` :/